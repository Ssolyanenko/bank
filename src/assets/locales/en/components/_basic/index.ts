import modalBox from './modalBox.json';
import cardBlock from './cardBlock.json';
import stepperForm from './stepperForm.json';

export const translationsBasic = {
  ...modalBox,
  ...cardBlock,
  ...stepperForm,
};
