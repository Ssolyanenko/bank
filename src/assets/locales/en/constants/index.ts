import menu from './menu.json';
import errors from './errors.json';
import transferListContent from './transferListContent.json';
import stepForm from './stepForm.json';
import buttonNames from './buttonNames.json';
import lockCardModalContent from './lockCardModalContent.json';
import user from './user.json';
import input from './input.json';
import activateBlockCardContent from './activateBlockCardContent.json';
import resetFormsModalsContent from './resetFormsModalsContent.json';

export const translationsConstants = {
  ...menu,
  ...errors,
  ...transferListContent,
  ...stepForm,
  ...buttonNames,
  ...lockCardModalContent,
  ...user,
  ...input,
  ...activateBlockCardContent,
  ...resetFormsModalsContent,
};
