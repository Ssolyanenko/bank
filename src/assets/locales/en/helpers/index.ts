import security from './security.json';

export const translationsHelpers = {
  ...security,
};
