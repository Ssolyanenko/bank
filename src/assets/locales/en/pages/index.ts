import landingPage from './landingPage.json';
import mainPageContent from './mainPageContent.json';
import notFoundPage from './notFoundPage.json';
import myCardsPage from './myCardsPage.json';

export const translationsPages = {
  ...landingPage,
  ...mainPageContent,
  ...notFoundPage,
  ...myCardsPage,
};
