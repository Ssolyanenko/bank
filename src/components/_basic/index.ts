export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
