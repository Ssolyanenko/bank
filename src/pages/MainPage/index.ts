export * from './MainPage';
export * from './MainPageContent';
