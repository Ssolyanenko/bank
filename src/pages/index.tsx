export * from 'pages/CreditCardsPage';
export * from 'pages/LandingPage';
export * from 'pages/MainPage';
export * from 'components/_basic/SecureWrapper';
export * from 'pages/MyCardApplicationsPage';
export * from 'pages/MyCardsPage';
export * from 'pages/CreditCardPremiumPage';
export * from 'pages/CreditCardBillablePage';
export * from 'pages/LayoutPage';
export * from 'pages/NotFoundPage';
