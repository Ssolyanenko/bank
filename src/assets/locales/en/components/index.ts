import { translationsMainPageComponents } from './mainPageComponents';
import { translationsForms } from './forms';
import { translationsExchageRates } from './exchangeRates';
import { translationsLandingPageComponents } from './landingPageComponents';
import { translationsMyCardApplications } from './myCardApplications';
import { translationsContacts } from './contacts';
import { translationsCardsLimits } from './transactionLimit';
import { translationsCardHistory } from './cardHistory';
import { translationsBasic } from './_basic';

export const translationsComponents = {
  ...translationsMainPageComponents,
  ...translationsForms,
  ...translationsExchageRates,
  ...translationsLandingPageComponents,
  ...translationsMyCardApplications,
  ...translationsContacts,
  ...translationsBasic,
  ...translationsCardHistory,
  ...translationsCardsLimits,
};
