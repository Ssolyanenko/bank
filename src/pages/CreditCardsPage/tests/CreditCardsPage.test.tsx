import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router';

import { ModalConfirmation } from 'components/_basic';
import { CreditCardsPage } from 'pages';
import { mockInitialState } from 'constants/mockInitialState';
import { mockStore } from 'helpers';

describe('<CreditCardsPage />', () => {
  const store = mockStore(mockInitialState);

  test('should check a snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Router>
          <CreditCardsPage />
        </Router>
      </Provider>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should check snapshot modal with initial state', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <CreditCardsPage />
      </Provider>
    );
    const component = wrapper.getElement();
    const modal = wrapper.find(ModalConfirmation);

    expect(component).not.toContain(modal);
  });

  test('should check snapshot component with isOpened modal', () => {
    const setHookState = (newState: { isOpened: boolean }) => jest.fn().mockImplementation(() => [newState, () => {}]);
    const reactMock = require('react');
    reactMock.useState = setHookState({ isOpened: true });

    const wrapper = shallow(
      <Provider store={store}>
        <CreditCardsPage />
      </Provider>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });
});
