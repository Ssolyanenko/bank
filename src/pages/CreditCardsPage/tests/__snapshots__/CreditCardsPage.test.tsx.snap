// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreditCardsPage /> should check a snapshot 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <CreditCardsPage />
  </MemoryRouter>
</Context.Provider>
`;

exports[`<CreditCardsPage /> should check snapshot component with isOpened modal 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <CreditCardsPage />
</Context.Provider>
`;
